pkg/costmodel/allocation.go

```diff
@@ -625,7 +621,7 @@ func applyPodResults(window kubecost.Window, resolution time.Duration, podMap ma
 
 func applyCPUCoresAllocated(podMap map[podKey]*Pod, resCPUCoresAllocated []*prom.QueryResult) {
 	for _, res := range resCPUCoresAllocated {
-		key, err := resultPodKey(res, env.GetPromClusterLabel(), "namespace", "pod")
```

Comment:
why are we removing pod here?

Reply:
Check the resultPodKey function below-- it's refactored to enforce the labels we're expecting to see, in the order we're expecting to see them, to take the logic out of the function caller.

Reply:
Ah yeah, I see